sky/backends/cloud_vm_ray_backend.py

```diff
@@ -1821,7 +1810,7 @@ def _ensure_cluster_ray_started(self, handle: 'CloudVmRayResourceHandle',
             # At this state, an erroneous cluster may not have cached
             # handle.head_ip (global_user_state.add_or_update_cluster(...,
             # ready=True)).
-            use_cached_head_ip=False)
+            use_cached_head_ip=None)
```

Comment:
There are a couple places in this PR where previously we always query the cloud for IPs, but with this PR we always try to use cached IPs first. Why is that?

Reply:
Looks like all callers now use None for this arg. Ok to eliminate the arg?

Reply:
Good point! Just removed the argument. Testing:
- [ ] `pytest tests/test_smoke.py`